web/src/narrow.js

```diff
@@ -541,6 +543,14 @@ export function activate(raw_operators, opts) {
 
         search.update_button_visibility();
 
+        // Clears 'compose_in_search_view' when manually exiting a
+        // search view narrow.
+        compose_banner.clear_in_search_view();
```

Comment:
Why did you need to do this here instead of where the rest of the warnings are cleared in `clear_warnings`?

Reply:
I didn't do it this way previously and got a bug where after `clear_warnings` was called, the `WARNING` CSS label was gone and couldn't be restored. For some reason, when I cleared the banner this way, I was still able to create a warning banner in the lines after. That's why I updated clear_warnings to have `.not(".compose_in_search_view").remove();`.

Weird bug. Not 100% sure why it worked that way, but it did.